test/unit/test_main.py

```diff
@@ -46,6 +46,7 @@ def setUp(self):
 
     def test_master_args_correctly_create_cluster_master(self):
         mock_cluster_master = self.mock_ClusterMaster.return_value  # get the mock for the ClusterMaster instance
+        mock_cluster_master._build_store = _FakeBuildStore()
```

Comment:
We need to manually add a _build_store because private member attributes aren't created by mock objects.

This is here because we call `master._build_store.clean_up()` when the master service shuts down -- we might be able to remove this later if we decide to move this logic somewhere else in the app